Assets/!/Scripts/PlayerAudio.cs

```diff
@@ -29,29 +43,108 @@ public void Crawl()
 
     public void Run()
     {
-        PlayAudio("Walk", 1f);
+        PlayRandomAudio(RandomClip.Run, 1f);
+    }
+
+    public void DieScream()
+    {
+        PlayRandomAudio(RandomClip.DieScream, 0.8f);
+    }
+
+    public void Blood()
+    {
+        PlayRandomAudio(RandomClip.Blood, 3.5f);
+    }
+
+    public void BulletHit()
+    {
+        PlayAudio("BulletHit", 1.5f);
     }
 
     public void Jump()
     {
         PlayAudio("Jump", 1f);
     }
 
+    public void FallDown()
+    {
+        PlayAudio("FallDown", 0.7f);
+    }
+
+    private void PlayRandomAudio(RandomClip clipType, float volume)
+    {
+        string clipName = RandomAudio(clipType);
+        PlayAudio(clipName, volume);
+    }
+
     public void PlayAudio(string clipName, float volume)
     {
-        Audio audioToPlay = System.Array.Find(audioPlayer, a => a.id == clipName);
+        Audio audioToPlay = FindAudioClip(clipName);
 
         if (audioToPlay.clip == null)
         {
             UnityEngine.Debug.LogWarning("Audio clip not found: " + clipName);
             return;
         }
 
-
         audioSource.clip = audioToPlay.clip;
         audioSource.volume = volume;
         audioSource.Play();
         audioSource.loop = false;
     }
 
+    private Audio FindAudioClip(string clipName)
+    {
+        Audio audioToPlay = System.Array.Find(walkAudio, a => a.id == clipName);
+        if (audioToPlay.clip == null)
+        {
+            audioToPlay = System.Array.Find(runAudio, a => a.id == clipName);
+        }
+        if (audioToPlay.clip == null)
+        {
+            audioToPlay = System.Array.Find(dieScreamAudio, a => a.id == clipName);
+        }
+        if (audioToPlay.clip == null)
+        {
+            audioToPlay = System.Array.Find(bloodAudio, a => a.id == clipName);
+        }
+        if (audioToPlay.clip == null)
+        {
+            audioToPlay = System.Array.Find(otherAudio, a => a.id == clipName);
+        }
```

Comment:
Defeats purpose of having different arrays, this is the same as having one big array